protocol/configurable-calldata-gas.md

```diff
@@ -0,0 +1,77 @@
+
+Since Pectra increased L1's data availability capacity to a target of 6 128kb blobs every 12 seconds, it can handle 64 kb/s of throughput. A single rollup with a two-second block time can only accept 128kb of batches per block. At 40 gas per byte (as of Pectra's [EIP 7623](https://eips.ethereum.org/EIPS/eip-7623)), it would cost 5.12M gas per block for a rollup's calldata to use enough blobs to hit L1's blob target. Most OP Stack chains have gas targets high enough to accept far more than 128kb of calldata per block! This allows blocks to exceed the rollup's constraints without putting any pressure on the base fee to price out this congestion.
+
+| Chain | Gas target | Underpricing factor |
+|-------|------------|---------------------|
+| Base | 50M | ≈ 9.8× |
+| OP Mainnet | 20M | ≈ 3.9× |
+| Ink  | 15M | ≈ 2.9× |
+| Unichain | 15M | ≈ 2.9× |
+| Mode | 15M | ≈ 2.9× | 
+
+Because so many OP Stack rollups have underpriced calldata, these chains rely on [batcher sequencer throttling](https://docs.optimism.io/operators/chain-operators/configuration/batcher#batcher-sequencer-throttling) to prevent large backlogs of batches from accumulating. The downside of this approach is that it "breaks the fee market." Users must outbid each other's priority fees in a first-price auction instead of expecting all transactions willing to pay the current base fee to be included in blocks. Many applications do not expect to actively participate in priority fee auctions because EIP 1559 has been so successful at eliminating them, so they experience these periods as a denial of service. (The base fee also plummets to zero during these periods, so even when the throttle is no longer binding, it takes time for the priority fee auction to end.)
+
+Pricing calldata correctly should make batcher sequencer throttling unnecessary and eliminate the priority fee auctions we see today when calldata is congested.
+
+---
+
+# Alternatives Considered
+
+* **Continue hard-coding gas costs**. Updating gas costs once or twice per year on a similar cadence to L1 hard forks is likely frequent enough for most chains, but it doesn't address differences in gas targets between OP Stack chains without each chain maintaining a persistent fork of the client.
+* **Full fee-abstraction** (see [`specs#73`](https://github.com/ethereum-optimism/specs/issues/73)). Only addresses fees, but we need to manipulate gas consumption for the fee market to function properly.
+* **Expose a single configuration value**. We don't foresee `eip7623StandardTokenCost` ever being modified from its default value of `4`, but it's included in the proposal for completeness.
+* **Change the actual calldata gas cost, not just the floor**. This diverges from how the L1 community has been addressing calldata costs and impacts more transactions than are necessary to solve the problem.
```

Comment:
This approach seems simpler and also more directly addresses the issue of L2 calldata. EIP-7623 was implemented with a floor function in order to discourage calldata-focused transactions, steering them towards blobs instead. If all L2 data must be batched to L1, then we may want to meter all calldata equally.

Reply:
I think the floor is actually better. It caps the amount of calldata that can be used in a block without reducing the amount of execution that can happen in a block.